src/service/permissions-service.ts

```diff
@@ -0,0 +1,101 @@
+import { roles } from '../constants/roles';
+import { UserRoles, FallbackRole } from '../constants/user-roles';
+import { jwtDecode } from 'jwt-decode';
+
+interface DecodedToken {
+  role: string;
+}
+
+export const PermissionsService = (() => {
+  /**
+   * This function takes a the JWT token of the user and a permission, and validates if the user is allowed to perform the action associated with the permission.
+   * @param {string} token A JWT token string from Firestore User.
+   * @param {string} permission The permission to check.
+   * @returns {Boolean} True if the user has the permission, false otherwise.
+   */
+  const canUser = (token: string, permission: string) => {
+    console.log(`[ROAR Permissions Manager] Checking permission ${permission} with token: ${token}`);
+    try {
+      const userRole = getRoleFromToken(token)?.toUpperCase();
+
+      // If the user is a super admin, grant permission.
+      if (userRole === UserRoles.SUPER_ADMIN) return true;
+
+      const config = roles[userRole as keyof typeof roles];
+
+      // If the user role doesn't exist in our config, flag and deny.
+      if (!config) {
+        console.error(`[ROAR Permissions Service] Invalid user role "${userRole}".`);
+        return false;
+      }
+
+      return checkPermissionList(config.permissions, permission);
+    } catch (error) {
+      console.error('[ROAR Permissions Service] Error checking permissions:', error);
+      return false;
+    }
+  };
+
+  /**
+   * This function takes a JWT token and returns the user's role. If the token is invalid or missing the role claim,
+   * return the GUEST role.
+   *
+   * @param {string} token JWT token string from Firestore User.
+   * @returns {string} The user's role based on the provided JWT token.
+   */
+  const getRoleFromToken = (token: string) => {
+    const decodedToken = jwtDecode<DecodedToken>(token);
+    const userRole = decodedToken.role ?? FallbackRole;
+
+    // Retrieve the user's role from the token's claims. If the claim is missing or invalid, default to the GUEST role.
+    if (!decodedToken.role) {
+      console.warn(
+        `[ROAR Permissions Manager] Invalid or missing role claim in User's custom claims. Defaulting to the ${FallbackRole} role.`,
+      );
```

Comment:
This is technically an error as all tokens should have a `role` claim. We should also use `userRole` in the log to ensure proper reporting back to Sentry when the `console.error` is logged.

```suggestion
      console.error(
        `[ROAR Permissions Service] Missing role in user's claims. Defaulting to the ${userRole} role.`,
      );
```